beacon_node/network/src/sync/range_sync/chain.rs

```diff
@@ -770,51 +756,25 @@ impl<T: BeaconChainTypes> SyncingChain<T> {
             return None;
         }
 
-        // One is added to the start slot to begin one slot after the epoch boundary
-        let batch_start_slot = self
-            .start_epoch
-            .start_slot(slots_per_epoch)
-            .saturating_add(1u64)
-            + self.to_be_downloaded_id.saturating_sub(1) * blocks_per_batch;
-
         // don't request batches beyond the target head slot
-        if batch_start_slot > self.target_head_slot {
+        if self.to_be_downloaded.start_slot(slots_per_epoch) > self.target_head_slot {
             return None;
         }
 
         // truncate the batch to the epoch containing the target head of the chain
         let batch_end_slot = std::cmp::min(
             // request either a batch containing the max number of blocks per batch
-            batch_start_slot + blocks_per_batch,
+            self.to_be_downloaded.start_slot(slots_per_epoch) + blocks_per_batch + 1,
```

Comment:
not sure if here this should be a `+2` @AgeManning

Reply:
This should be a 1. We are shifting the start_slot by 1 so this +1 is for that start slot, then we do blocks_per_batch on top. 
This looks correct to me